### PR TITLE
Adding the Wagon class.

### PR DESCRIPTION
**Pull request by Silk8Worm**
I added a wagon class that is tradeable and driveable. This happens to be a super wagon with an engine, so don't question why it's driveable. However, you can't ride it because the engine takes up all the space.

### CODE REVIEW

File: src/main/java/Wagon.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/* A wagon class that is drivable and treatable.
```
Comment (marianmwang) [inline src/main/java/Wagon.java]: typo, it should say "tradable"

File: src/test/java/WagonTest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
+/*
+ * This file contains sample JUnit test cases for Horse.java
```
Comment (marianmwang) [inline src/test/java/WagonTest.java]: this is not for Horse.java, this is for Wagon.java

**Review by marianmwang — CHANGES_REQUESTED** [review]
found 2 typos

File: src/main/java/Wagon.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        this.maxSpeed += 10;
+    }
+
+    @Override
+    public void downgradeSpeed() {
+        this.maxSpeed -= 10;
+    }
+
+    @Override
+    // This method returns the maximum speed taken from the super class, Trader. The max speed is assigned during the construction of this class.
```
Comment (marianmwang) [inline src/main/java/Wagon.java]: ??????????????????

File: src/main/java/Wagon.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    @Override
+    public int getPrice() {
+        return 400;
+    }
+
+    @Override
+    public String toString(){
+        String oof = super.toString();
```
Comment (marianmwang) [inline src/main/java/Wagon.java]: bad naming convention

**Review by marianmwang — CHANGES_REQUESTED** [review]
comment errors and variable naming

File: src/main/java/Wagon.java (hunk 4 above)
Comment (NathanZC) [reply src/main/java/Wagon.java]: Quality name

**Pull request merged**